src/open_inwoner/userfeed/hooks/external_task.py

```diff
@@ -0,0 +1,88 @@
+from typing import List
+
+from django.utils.translation import gettext_lazy as _
+
+from open_inwoner.accounts.choices import LoginTypeChoices
+from open_inwoner.accounts.models import User
+from open_inwoner.openzaak.api_models import OpenTask
+from open_inwoner.openzaak.clients import build_client
+from open_inwoner.userfeed.adapter import FeedItem
+from open_inwoner.userfeed.choices import FeedItemType
+from open_inwoner.userfeed.models import FeedItemData
+
+from ..adapters import register_item_adapter
+
+
+class OpenTaskFeedItem(FeedItem):
+    base_title = _("Open task")
+    base_message = _("Open task that is yet to be completed")
+
+    @property
+    def title(self) -> str:
+        return f"{self.base_title} ({self.get_data('task_identificatie')})"
+
+    @property
+    def message(self) -> str:
+        return self.get_data("task_name", super().message)
+
+
+def update_external_task_items(user: User, openstaande_taken: List[OpenTask]):
+    """
+    Creates items for OpenTasks if they do not exist yet, updates existing items if the
+    data changed and marks existing items as complete if no OpenTask exists for that
+    uuid anymore
+    """
+    existing_uuid_mapping = {
+        str(item.ref_uuid): item
+        for item in FeedItemData.objects.filter(
+            type=FeedItemType.external_task,
+            user=user,
+        )
+    }
+    existing_uuids = set(existing_uuid_mapping.keys())
+
+    update_data = []
+    create_data = []
+    for task in openstaande_taken:
+        type_data = dict(
+            action_url=task.formulier_link,
+            task_name=task.naam,
+            task_identificatie=task.identificatie,
+        )
```

Comment:
I would make this a dict literal instead of a dict constructor so it is clear at a glance these are arbitrary keys. We want them highly visible because they are important.